Sources/Workspace/Workspace.swift

```diff
@@ -255,24 +247,51 @@ public class Workspace {
     /// - Parameters:
     ///   - fileSystem: The file system to use.
     ///   - location: Workspace location configuration.
-    ///   - mirrors: Dependencies mirrors.
-    ///   - authorizationProvider: Provider of authentication information.
-    ///   - customToolsVersion: A custom tools version.
-    ///   - customManifestLoader: A custom manifest loader.
-    ///   - customRepositoryManager: A custom repository manager.
-    ///   - customRepositoryProvider: A custom repository provider.
-    ///   - customIdentityResolver: A custom identity resolver.
-    ///   - customHTTPClient: A custom http client.
-    ///   - customArchiver: A custom archiver.
-    ///   - customChecksumAlgorithm: A custom checksum algorithm.
-    ///   - customFingerprintStorage: A custom fingerprint storage.
-    ///   - additionalFileRules: File rules to determine resource handling behavior.
-    ///   - resolverUpdateEnabled: Enables the dependencies resolver automatic version update check.  Enabled by default. When disabled the resolver relies only on the resolved version file
-    ///   - resolverPrefetchingEnabled: Enables the dependencies resolver prefetching based on the resolved version file.  Enabled by default.
-    ///   - resolverFingerprintCheckingMode: Fingerprint checking mode. Defaults to `.warn`.
-    ///   - sharedRepositoriesCacheEnabled: Enables the shared repository cache. Enabled by default.
+    ///   - registries: Configuration for registries
+    ///   - authorizationProvider: Provider of authentication information for outbound network requests.
+    ///   - resolutionConfiguration: Configuration to fine tune the dependency resolution behavior.
+    ///   - customManifestLoader: Custom manifest loader. Used to customize how manifest are loaded.
+    ///   - customPackageContainerProvider: Custom package container provider. Used to provide specialized package providers.
+    ///   - customRepositoryProvider: Custom repository provider. Used to customize source control access.
     ///   - delegate: Delegate for workspace events
-    public init(
+    public convenience init(
+        fileSystem: FileSystem,
+        location: Location,
+        authorizationProvider: AuthorizationProvider? = .none,
+        resolutionConfiguration: ResolutionConfiguration? = .none,
+        // optional customization, primarily designed for testing
+        customManifestLoader: ManifestLoaderProtocol? = .none,
+        customPackageContainerProvider: PackageContainerProvider? = .none,
+        customRepositoryProvider: RepositoryProvider? = .none,
+        // delegate
+        delegate: WorkspaceDelegate? = .none
+    ) throws {
```

Comment:
this is the key change in this PR:
=> this would be the new public initializer, offering only a handful of customization options
=> tests will use an internal initializer that offers many more customization options

Reply:
Looks good to me.  Do we still need the word "custom", though?  Why do `authorizationProvider` and `resolutionConfiguration` not have it but the other three do?  All of the options seem to have a default of `.none`, which presumably mean the default one.  So we could just drop "custom" IMO.

Reply:
The difference is that authorizationProvider does not have a default and resolutionConfiguration is settings / preferences that we expect the call site to define based on user preference like CLI flags, env variables, user defaults, etc. The rest of the arguments are extension points, where the there is a default that should cover most of the use cases and only more advanced use cases are expected to customize them, hence the "custom" prefix to emphasize that.